cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/DumpCommandBase.java

```diff
@@ -0,0 +1,51 @@
+package cloud.foundry.cli.services;
+
+import static picocli.CommandLine.Mixin;
+
+import cloud.foundry.cli.crosscutting.logging.Log;
+import cloud.foundry.cli.crosscutting.mapping.YamlMapper;
+import cloud.foundry.cli.crosscutting.mapping.beans.Bean;
+
+import java.util.concurrent.Callable;
+
+/**
+ * The base class for concrete dump commands. It implements the process of resolving and interpreting the provided
+ * configuration file via a {@link Callable<Integer>} interface (as any other command class).
+ * @param <B> the bean type in which the resolved config will be interpreted
+ */
+public abstract class DumpCommandBase<B extends Bean> implements Callable<Integer> {
```

Comment:
Mind to explain why we use a generic class rather than just using the `Bean` interface below? I'm pretty sure it's got something to do with the dumping not working as intended then, but this should be explained at least inside the class (doesn't need to be in a docstring necessarily).

Reply:
By doing that, I tried to reduce duplicate code in the DumpController class. In other controller classes we have lots of duplicate code.